Indicators/CorrelationIndicator.cs

```diff
@@ -0,0 +1,211 @@
+        /// and period values
+        /// </summary>
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        /// <param name="correlationType">Correlation type</param>
+        public CorrelationIndicator(Symbol targetSymbol, Symbol referenceSymbol, int period, CorrelationIndicatorType correlationType = CorrelationIndicatorType.Pearson)
+            : this($"Correlation({period})", targetSymbol, referenceSymbol, period, correlationType)
+        {
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// 
+        /// As this indicator is receiving data points from two different symbols,
+        /// it's going to compute the next value when the amount of data points
+        /// of each of them is the same. Otherwise, it will return the last correlation
+        /// value computed
+        /// </summary>
+        /// <param name="input">The input value of this indicator on this time step.
+        /// It can be either from the target or the reference symbol</param>
+        /// <returns>The correlation value of the target used in relation with the reference</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            var inputSymbol = input.Symbol;
+            if (inputSymbol == _targetSymbol)
+            {
+                _targetDataPoints.Add(input.Close);
+            } 
+            else if(inputSymbol == _referenceSymbol)
+            {
+                _referenceDataPoints.Add(input.Close);
+            }
+            else
+            {
+                throw new ArgumentException("The given symbol was not target or reference symbol");
+            }
+
+            if (_targetDataPoints.Samples == _referenceDataPoints.Samples && _referenceDataPoints.Count > 1)
+            {
+                ComputeCorrelation();
+            }
+            return _correlation;
+        }
+
+        /// <summary>
+        /// Computes the returns with the new given data point and the last given data point
+        /// </summary>
+        /// <param name="rollingWindow">The collection of data points from which we want
+        /// to compute the return</param>
+        /// <returns>The returns with the new given data point</returns>
+        private static decimal GetNewReturn(RollingWindow<decimal> rollingWindow)
+        {
+            return ((rollingWindow[0] / rollingWindow[1]) - 1);
+        }
+
+        /// <summary>
+        /// Computes the correlation value usuing symbols values
+        /// correlation values assing into _correlation property
+        /// </summary>
+        private void ComputeCorrelation()
+        {
+            if (_targetDataPoints.Count < _period || _referenceDataPoints.Count < _period)
```

Comment:
The called is also checking other cases where the correlation might no need to be calculated `_targetDataPoints.Samples == _referenceDataPoints.Samples && _referenceDataPoints.Count > 1`, believe these checks should be merged into a single location